src/TBuffer.cpp

```diff
@@ -30,6 +29,552 @@
 
 #include <assert.h>
 
+// Define this to get qDebug() messages about the decoding of UTF-8 data when it
+// is not the single bytes of pure ASCII text:
+#define DEBUG_UTF8_PROCESSING
```

Comment:
Uh, yeah!  Reports on __every__ multi-byte UTF-8 sequence so WILL be spammy but will also be useful when debugging - I am going to ensure this is defined when I look more closely into that broken Cyrillic MUD replay... 😀